lib/active_record_doctor/detectors/incorrect_length_validation.rb

```diff
@@ -55,6 +57,20 @@ def maximum_allowed_by_validations(model, column)
         end
         length_validator ? length_validator.options[:maximum] : nil
       end
+
+      def column_limit(table, column)
+        return column.limit if column.limit
+
+        # Example: char_length(name::text) <= 64
+        pattern = /(char_|character_)?length\(['"`]?#{column.name}(::text)?['"`]?\)\s*<=\s*(?<limit>\d+)/i
```

Comment:
I suggest we add support for both `<` and `<=` and adjust the limit accordingly, i.e. `< 100` means the limit is `<= 99`.